lib/server.go

```diff
@@ -2468,18 +2485,24 @@ func (srv *Server) _startAddressRelayer() {
 		if atomic.LoadInt32(&srv.shutdown) >= 1 {
 			break
 		}
-		// For the first ten minutes after the server starts, relay our address to all
+		// For the first ten minutes after the connection controller starts, relay our address to all
 		// peers. After the first ten minutes, do it once every 24 hours.
-		// TODO: temporary
-		addressMgr := addrmgr.New("", net.LookupIP)
-		glog.V(1).Infof("Server.Start._startAddressRelayer: Relaying our own addr to peers")
+		glog.V(1).Infof("Server.startAddressRelayer: Relaying our own addr to peers")
+		remoteNodes := srv.connectionController.rnManager.GetAllRemoteNodes().GetAll()
 		if numMinutesPassed < 10 || numMinutesPassed%(RebroadcastNodeAddrIntervalMinutes) == 0 {
-			for _, pp := range srv.cmgr.GetAllPeers() {
-				bestAddress := addressMgr.GetBestLocalAddress(pp.netAddr)
+			for _, rn := range remoteNodes {
+				if !rn.IsHandshakeCompleted() {
+					continue
+				}
+				netAddr := rn.GetNetAddress()
+				if netAddr == nil {
+					continue
+				}
+				bestAddress := srv.AddrMgr.GetBestLocalAddress(netAddr)
 				if bestAddress != nil {
-					glog.V(2).Infof("Server.Start._startAddressRelayer: Relaying address %v to "+
-						"peer %v", bestAddress.IP.String(), pp)
-					pp.AddDeSoMessage(&MsgDeSoAddr{
+					glog.V(2).Infof("Server.startAddressRelayer: Relaying address %v to "+
+						"RemoteNode (id= %v)", bestAddress.IP.String(), rn.GetId())
+					if err := rn.SendMessage(&MsgDeSoAddr{
```

Comment:
nit: clean up nesting for readability
```suggestion
					addrMsg := MsgDeSoAddr{
```